### DISCUSSION TITLE
VNet - pomoc/ polecenie kursu online

### DISCUSSION THREAD

**Discussion by foxwsp** [AzureCommunityPL/group#98] [Pomoc] [answered]
Hej,

Nigdy specjalnie nie zajmowalem sie networkingiem w Azure az do teraz.
Mam problem, poniewaz nie do konca wiem jak dzialaja subnety i na jakich zasadach moja aplikacja w subnecie A bedzie, a moze nie bedzie sie widziala z VM w subnecie B. 

Znalazlem na pluralsight kurs: Managing Microsoft Azure Virtual Networks, ale idzie to jakos powoli.
Czy moze ktos z was wie czy "zasoby" normalnie sie widza miedzy subnetami w jednym vnecie?

**Comment by kaluzaaa** [comment]
> Czy moze ktos z was wie czy "zasoby" normalnie sie widza miedzy subnetami w jednym vnecie?

Tak działa to domyślnie, jeśli nie masz żadnego NSG (network security group)

Co do nauki to:

1. https://docs.microsoft.com/en-us/learn/modules/intro-to-azure-networking/
2. https://docs.microsoft.com/en-us/learn/paths/architect-network-infrastructure/
3. Przykłady z dokumentacji! Serio!

**Reply by kamil-mrzyglod** [reply]
I pewnie najłatwiej na podstawie dokumentacji "łopatologicznie" zobaczyć jak się sprawy mają 😉

**Reply by KamilWiecek** [reply]
Wewnątrz VNETu by default wszystko się widzi. Domyślne reguły w NSG dalej pozwolą na ruch pomiędzy subnetami a nawet więcej:
```
The virtual network address space (all IP address ranges defined for the virtual network), all connected on-premises address spaces, peered virtual networks, virtual networks connected to a virtual network gateway, the virtual IP address of the host, and address prefixes used on user-defined routes. This tag might also contain default routes.
```

Dopiero jak dodasz NSG rule, która to zablokuje, to możesz zapewnić, że workloady nie będą się widzieć. Musisz to zrobić intencjonalnie.

Wszystko znajdziesz tak jak napisali koledzy wcześniej w dokumentacji, ja zacząłbym tutaj https://docs.microsoft.com/en-us/azure/architecture/browse/#networking, bo po wejśćiu w określoną architekturę masz jej opis.

**Comment by foxwsp** [comment]
Czytałem już oba, i dalej nie jestem w stanie rozwiązać mojego problemu.
Szukam na stacku i widzę ze nie ja jedyny mam taki problem: https://stackoverflow.com/questions/34998899/vnet-between-virtual-machine-and-app-service-in-azure

Postaram się go opisać dokładniej:

Posiadam 2 aplikacje App Service + VMkę z Ubuntu 18.04.
Te 2 aplikacje są wystawione pod publicznymi adresami i są w subnecie X, a jedna z nich musi dostać się do tej maszyny wirtualnej w subnecie Y - oczywiście oba subnety są w jednym Vnecie. 

Na VMce mam RabbitMQ, więc w inbound rules otwarłem port rabbita w ustawieniach VM oraz nadałem jej statyczne IP w sieci prywatnej. Niestety ilekroć moja aplikacja stara się dobić do Rabbita to jej się to nie udaje.

**Reply by kamil-mrzyglod** [reply]
Czyli jak rozumiem masz włączone VNet integration? Możesz pokazać swoje ustawienia NSG?

**Reply by KamilWiecek** [reply]
JW - jeżeli VNnet integration jest skonfigurowane, to użyj Kudu, żeby przetestować połączenie. Upewniej się, że firewall na poziomie OS VM-ki z Rabbitem też jest dobrze skonfigurowany.

**Reply by eklime** [reply]
Czekaj czekaj - jak ustawiałeś statyczny adres IP?

**Reply by foxwsp** [reply]
@kamil-mrzyglod nie mam żadnego nsg w całej topologii:
![image](https://user-images.githubusercontent.com/10991279/95292293-7dcf5b00-0871-11eb-99ff-867b5f22af68.png)

@kwiecek sprawdziłem porty na VMce - jest ok.

@eklime na network interfejsie ze screena topologii (vndrabbit179) nadałem że publiczne IP ma stałe prywatne, ale to już usunąłem.

**Reply by foxwsp** [reply]
dodam jeszcze, że te moje 2 aplikacje App Serivce są w public subnet i faktycznie ta o którą mi chodzi mi integration włączone.

**Comment by foxwsp** [comment, marked as answer]
Udało się, jestem retardem i mialem zle ustawione virtual hosty na Rabbicie ;(
Przepraszam za całe zamieszanie i dziękuję za pomoc.
